entry.go

```diff
@@ -29,6 +29,10 @@ type Entry struct {
 
 	// Message passed to Debug, Info, Warn, Error, Fatal or Panic
 	Message string
+
+	// A receive-only notification channel for knowing when all hooks have
+	// finished firing.
+	HooksDone <-chan struct{}
```

Comment:
The main purpose was to ensure it could be tested properly, so it could easily be made hidden and used only for testing.